spec/message.abnf

```diff
@@ -42,11 +42,15 @@ unquoted = unquoted-start *name-char
 unquoted-start = name-start / DIGIT / "."
                / %xB7 / %x300-36F / %x203F-2040
 
-; reserve additional sigils for use by future versions of this
-; specification or for private use by implementations
-reserved       = ( reserved-start / private-start ) reserved-body
-reserved-start = "!" / "@" / "#" / "%" / "*" / "<" / ">" / "/" / "?" / "~"
+
+; reserve sigils for private-use by implementations
+private-use    = private-start 1*(text)
```

Comment:
It should be `*(text)` as you note.

I think the bug is in `reserved` 😄 

From your comment, I suspect you and I are thinking differently about the reserved/private constructs. Let me see if I can clarify.

The way I wrote `private-use` above is that parsers do not parse anything within a private use placeholder (except for the optional operand). It is up to the private use implementation to define what, if any, structure appears in the private use "blob". A smart implementer would probably define their private use syntax to be consistent with the rest of MF... but, hey, it's their funeral.

I think it makes sense to define `reserved` in the same way and we discussed making it utterly opaque to allow for innovations in syntax in the future. I want to send a note to our future selves that, if they unreserve a sigil, it SHOULD (2119 keyword intentional) probably work just like the rest of the syntax in terms of parsing. But those folks might ignore us.

I understood your point in an earlier comment to suggest that reserved sigils continue to parse word tokens, which is why you want to include `literal` in the production. That way if a sigil is unreserved "it just works". But I would tend to say that an newly-unreserved sigil would "just work" because it would then be part of `function` or some other new ABNF.

Functionally, using `*text` would set up our syntax so that once an implementation sees a `reserved` or (unknown) `private-use` sigil, they can consume the entire placeholder up to the (not escaped) `}` (and potentially emit the error we defined in formatting.md).

Reply:
> I understood your point in an earlier comment to suggest that reserved sigils continue to parse word tokens, which is why you want to include `literal` in the production. That way if a sigil is unreserved "it just works". But I would tend to say that an newly-unreserved sigil would "just work" because it would then be part of `function` or some other new ABNF.

My point is that if we use `*text`, we will not be able to do that later without breaking changes.

A key reason we're introducing `reserved` is to ensure that something like MF 2.1 will be able to take e.g. `!` and allow for its use as a `function` introducer. If we do that, we allow for a valid MF 2.1 expression like
```
{!new key=|no } escape|}
```
where the `|no } escape|` is a valid `quoted` value.

If we now define `reserved` as e.g.
```abnf
reserved = reserved-start *text
```
then it would mean that parsing an MF 2.1 message with an MF 2.0 processor could produce a syntax error.

Therefore the need to explicitly allow for `quoted` in `reserved-body`: If it's not there to start with, it can never be added later to any construct that we un-reserve.

For an alternative way to avoid this issue, we could require `{}` to always be escaped as `\{` and `\}` in `quoted`.

Reply:
I see it now.

It would be odd to require quoting `{`/`}` in a literal (since that would be different than quoting regimes in any programming language I can think of). Since `reserved` isn't allowed and `private-use` is discouraged (but not deprecated), most users will encounter this as a quirk when writing a literal.

Alright, you've convinced me.